cmd/upload-s3/upload-s3.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/evergreen-ci/pail"
+	"github.com/evergreen-ci/utility"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/level"
+	"github.com/mongodb/grip/send"
+	"github.com/pkg/errors"
+)
+
+// Upload a directory/file to S3.
+func main() {
+	var bucketName, local, remote, exclude string
+	flag.StringVar(&bucketName, "bucket", "", "the S3 bucket name")
+	flag.StringVar(&local, "local", "", "the local path")
+	flag.StringVar(&remote, "remote", "", "the remote path")
+	flag.StringVar(&exclude, "exclude", "", "file pattern to exclude from  upload")
+	flag.Parse()
+	if bucketName == "" {
+		grip.EmergencyFatal("bucket name is missing")
+	}
+	if local == "" {
+		grip.EmergencyFatal("local path is missing")
+	}
+	if remote == "" {
+		grip.EmergencyFatal("remote path is missing")
+	}
+
+	awsKey := os.Getenv("AWS_KEY")
+	if awsKey == "" {
+		grip.EmergencyFatal("AWS key is missing (env var: AWS_KEY)")
+	}
+	awsSecret := os.Getenv("AWS_SECRET")
+	if awsSecret == "" {
+		grip.EmergencyFatal("AWS secret is missing (env var: AWS_SECRET)")
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	client := utility.GetDefaultHTTPRetryableClient()
+	defer utility.PutHTTPClient(client)
+
+	opts := pail.S3Options{
+		Credentials: pail.CreateAWSCredentials(awsKey, awsSecret, ""),
+		Region:      endpoints.UsEast1RegionID,
+		Name:        bucketName,
+		MaxRetries:  10,
+		Permissions: pail.S3PermissionsPublicRead,
+		Verbose:     true,
+	}
+	bucket, err := pail.NewS3Bucket(opts)
+	if err != nil {
+		grip.EmergencyFatal("could not initialize bucket")
+	}
+
+	if err := grip.SetLevel(send.LevelInfo{Threshold: level.Debug, Default: level.Debug}); err != nil {
+		grip.EmergencyFatal(errors.Wrap(err, "setting grip logging level"))
+	}
+
+	if err := bucket.Push(ctx, pail.SyncOptions{
```

Comment:
Can't you just wrap the error and get rid of the if statement? Or does that not work with `grip.EmergencyFatal`?

Reply:
I could, but I find it harder to read when you have too many layers of function calls in one line like this: `grip.EmergencyFatal(errors.Wrapf(bucket.Push(...), "push failed in %s", args...))`